aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -567,16 +568,15 @@ module aptos_framework::transaction_validation {
 
             if (transaction_fee_amount > storage_fee_refunded) {
                 let burn_amount = transaction_fee_amount - storage_fee_refunded;
-                transaction_fee::burn_fee(gas_payer, burn_amount);
+                transaction_fee::burn_fee(gas_payer, burn_amount, account_addr != gas_payer);
             } else if (transaction_fee_amount < storage_fee_refunded) {
                 let mint_amount = storage_fee_refunded - transaction_fee_amount;
-                transaction_fee::mint_and_refund(gas_payer, mint_amount);
+                transaction_fee::mint_and_refund(gas_payer, mint_amount, account_addr != gas_payer);
```

Comment:
This would change any gas payer.  So, it could change accounts that you don't necessarily want to change to concurrent.

This should be feature flagged.